### PR TITLE
SDL: err-memory-handling

### PR DESCRIPTION
**Pull request by dmitry741**
SDL error memory handling

### CODE REVIEW

**Comment by SmirnovEgorRu** [comment]
make test

**Comment by SmirnovEgorRu** [comment]
Similar errors with clean code in PCM

File: algorithms/kernel/stump/stump_train_impl.i [hunk 1]
```diff
@@ -473,6 +473,7 @@ services::Status StumpTrainKernel<method, algorithmFPtype, cpu>::doStumpRegressi
     daal::tls<TGroupRes *> tls( [ = ]()-> TGroupRes *
     {
         TGroupRes *g = new TGroupRes();
+        DAAL_CHECK_STATUS_VAR(g)
```
Comment (ShvetsKS) [inline algorithms/kernel/stump/stump_train_impl.i]: why DAAL_CHECK_MALLOC is not used here?

Comment (dmitry741) [reply algorithms/kernel/stump/stump_train_impl.i]: It's lambda expression. It has to return TGroupRes pointer, not Status.

**Comment by ShvetsKS** [comment]
make test

**Comment by dmitry741** [comment]
restart

**Comment by ShvetsKS** [comment]
make test

**Pull request merged**